### PR TITLE
fix bug  #23044

### LINKED ISSUE #23044: [Bug]: Inconsistent `axis` Normalization in `ivy.gather` Across Backends
### Bug Explanation

The current implementation of the `gather` function in Ivy normalizes the axis parameter differently than native implementations in frameworks like `TensorFlow` , `PyTorch` or `Paddle`. Specifically, the line `axis = axis % len(params.shape)` may not be in line with how` TensorFlow` , `PyTorch` or `Paddle` handle `axis` normalization.

### Steps to Reproduce Bug

we're setting the value of `axis` to be: `axis = axis % len(params.shape)` across all backend at `ivy`
For `jax` backend:   https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/jax/general.py#L123

For `torch` backend: https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/torch/general.py#L174
For `tensorflow` backend: https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/tensorflow/general.py#L103
For `NumPy` backend:https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/numpy/general.py#L75
For `Paddle` backend:https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/paddle/general.py#L149

Now, this is how  `axis` value is being set at various framework native APIs:

for `torch`:
At `Pytorch` [gather](https://github.com/pytorch/pytorch/blob/4146be192ead477360a2763c5005e46a9485c3bf/torch/_inductor/lowering.py#L2472) we use a helper called : [_validate_dim](https://github.com/pytorch/pytorch/blob/4146be192ead477360a2763c5005e46a9485c3bf/torch/_inductor/lowering.py#L1100C4-L1100C18)

for `TensorFlow`:
for [tf.gather](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/ragged/ragged_gather_ops.py#L99) we use a helper called [get_positive_axis](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L6659)

For [paddle.gather](https://github.com/PaddlePaddle/Paddle/blob/release/2.5/python/paddle/tensor/manipulation.py#L2721) we set `axis = 0` 

### Environment

Linux + Docker + vscode

### Ivy Version

1.1.9

### Backend

- [X] NumPy
- [X] TensorFlow
- [X] PyTorch
- [X] JAX

### Device

CPU

### PR DESCRIPTION
**Pull request by Aryan8912**
# PR Description

## Related Issue

Close #23044 

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Sarvesh-Kesharwani** [comment]
Hi @Aryan8912,
would you mind resolving the merge conflicts please.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/backends/jax/general.py [hunk 1]
```diff
@@ -129,7 +152,7 @@ def gather(
     batch_dims: int = 0,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    axis = axis % len(params.shape)
+    axis = get_positive_axis_for_gather(axis, params.ndim)
```
Comment (Sarvesh-Kesharwani) [inline ivy/functional/backends/jax/general.py:155]: Hi @Aryan8912,
can you please describe what defect does this fix?

**Review by Sarvesh-Kesharwani — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
